From: Deniz <vrdons@proton.me>
Date: Wed, 29 Oct 2025 09:19:27 +0000
Subject: [PATCH 1/3] BACKPORT: seccomp: add filter_count field for KernelSU-Next

KernelSU requires struct seccomp to have atomic_t filter_count, but this field is missing on NON-GKI kernels.

MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Adding the field fixes build errors such as:

 error: no member named 'filter_count' in 'struct seccomp'

Reference: https://github.com/tiann/KernelSU/pull/2708
Actual Commit: https://github.com/GrapheneOS-Archive/kernel_common-5.10/commit/c818c03

Signed-off-by: Kees Cook <keescook@chromium.org>
Signed-off-by: Vrdons <vrdons@proton.me>
---
 include/linux/seccomp.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/seccomp.h b/include/linux/seccomp.h
index a9d5c52de4ea..4adb3b100d70 100644
--- a/include/linux/seccomp.h
+++ b/include/linux/seccomp.h
@@ -11,6 +11,7 @@
 #ifdef CONFIG_SECCOMP
 
 #include <linux/thread_info.h>
+#include <linux/atomic.h>
 #include <asm/seccomp.h>
 
 struct seccomp_filter;
@@ -27,6 +28,7 @@ struct seccomp_filter;
  */
 struct seccomp {
 	int mode;
+	atomic_t filter_count;
 	struct seccomp_filter *filter;
 };
 
-- 
2.50.1

